src/Razor/Microsoft.AspNetCore.Razor.Language/src/DefaultRazorTagHelperBinderPhase.cs

```diff
@@ -350,13 +352,13 @@ public override void VisitRazorDirective(RazorDirectiveSyntax node)
 
             internal static bool IsTypeInNamespace(string typeName, string @namespace)
             {
-                if (!TrySplitNamespaceAndType(typeName, out var typeNamespace, out var _) || typeNamespace == string.Empty)
+                if (!TrySplitNamespaceAndType(typeName, out var typeNamespace, out var _) || typeNamespace.Length == 0)
                 {
                     // Either the typeName is not the full type name or this type is at the top level.
                     return true;
                 }
 
-                return typeNamespace.Equals(@namespace, StringComparison.Ordinal);
+                return @namespace.Length == typeNamespace.Length && 0 == string.CompareOrdinal(typeName, typeNamespace.Start, @namespace, 0, @namespace.Length);
```

Comment:
I thought string.Equals would internally have a length check short circuit. What does doing the above get us?

Reply:
Oh I see what this is doing. Nvm 👍

Reply:
It prevents us from needing to allocate typeNamespace.

---
In reply to: [439618616](https://github.com/dotnet/aspnetcore/pull/22887#discussion_r439618616) [](ancestors = 439618616)